otelcol/config.go

```diff
@@ -131,3 +152,8 @@ func (cfg *Config) Validate() error {
 
 	return nil
 }
+
+// ToStringMap returns the map used to create this Config
+func (cfg *Config) ToStringMap() map[string]interface{} {
```

Comment:
Providing the map representation of the Config struct from the struct itself from seems like the best solution here. I made this a method to differentiate it from the other data properties on the struct, which then necessitated making a constructor for the struct to keep the field private.